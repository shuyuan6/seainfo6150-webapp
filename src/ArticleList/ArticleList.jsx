import React, { useEffect, useState } from "react";
import ArticleListItem from "../ArticleListItem/ArticleListItem.jsx"
import { isEmpty } from "lodash";

const ArticleList = (props) => {
  const [fetchedData, setFetchedData] = useState({});

  useEffect(() => {
    const fetchData = async () => {
      const response = await fetch(
        "http://demo1390455.mockable.io/articles"
      );
      const responseJson = await response.json();
      setFetchedData(responseJson);
    };

    if (isEmpty(fetchedData)) {
      fetchData();
    }
  }, [fetchedData]);

  let displayContent;

  console.log(typeof fetchedData);

  let vs = Object.values(fetchedData);
  
  if (!isEmpty(fetchedData)) {
    displayContent = (
      <ul>
        {vs.map((item) => <li key={item.slug}> {ArticleListItem(item)} </li>)}
      </ul>
    );
    
  } else {
    displayContent = <div><h1>You have no data!</h1></div>;
  }
  
  return (
    <section>
      {displayContent}
    </section>
  );
};

export default ArticleList;