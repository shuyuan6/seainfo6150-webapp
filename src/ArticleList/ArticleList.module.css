.container {
  background: #999;
  border: 1px dashed #666;
  border-radius: 10px;
  box-shadow: 0px 0px 10px 5px rgba(204, 0, 0, 0.5);
  display: grid;
  grid-template-columns: repeat(3, 1fr);
  gap: 20px;
  font-family: Arial, Helvetica, sans-serif;
  list-style-type: none;
  padding: 15px;
}

@media screen and (max-width: 800px) {
  .container {
    grid-template-columns: 1fr;
  }
}

@media screen and (min-width: 1600px) {
  .container {
    grid-template-columns: repeat(4, 1fr);
  }
}
