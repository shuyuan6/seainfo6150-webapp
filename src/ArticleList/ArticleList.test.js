import React from "react";
import ArticleList from "./ArticleList";

describe("ArticleList tests", () => {
  it("renders correctly", () => {
    const articles1 = [
       {
            title : "A title",
            timeStamp: "1919-01-01",
            displayDate: "1919-01-02",
            slug: "slug1",
            
        },
        {
            title : "A title",
            timeStamp: "1919-01-01",
            displayDate: "1919-01-02",
            slug: "slug2",
            
        }
    ]
    const { container } = render(<ArticleList articles={articles1}/>);
    expect(container).toMatchSnapshot();
  });
});
