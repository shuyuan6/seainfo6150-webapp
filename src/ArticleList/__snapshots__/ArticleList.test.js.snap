// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList tests renders correctly 1`] = `
<div>
  <div>
    <ul
      class="container"
    >
      <li
        class="container"
      >
        <article
          class="article"
        >
          <div
            class="wrapper"
          >
            <h2
              class="title"
            >
              A title
            </h2>
          </div>
          <button
            class="button"
          >
            Show more
          </button>
        </article>
      </li>
      <li
        class="container"
      >
        <article
          class="article"
        >
          <div
            class="wrapper"
          >
            <h2
              class="title"
            >
              A title
            </h2>
          </div>
          <button
            class="button"
          >
            Show more
          </button>
        </article>
      </li>
    </ul>
  </div>
</div>
`;
